### PR TITLE
separate glyph per unique path in SVGs

### PR DESCRIPTION
**Pull request by das-g**
### Reviewed by
- [x] @hixi
- [ ] @beneditatan

### CODE REVIEW

**Comment by das-g** [comment]
~~Note that SVG path elements that are almost but not exactly the same will still generate separate glyphs. (Thus the double airplanes, trains etc in the Font.)~~

fixed with f14748e

File: fontforge_font_creator/creator.py [hunk 1]
```diff
@@ -36,8 +46,12 @@ def create_font(self):
             self.fontforge_font.weight = 'normal'
             for hex_position, glyph in definition['mappings'].items():
                 hex_value = int(hex_position, 16)
-                print(hex_value, glyph['filename'])
-                self.add_glyph(hex_value, glyph['filename'])
+                print(hex_value, glyph['filename'], glyph['element'])
```
Comment (hixi) [inline fontforge_font_creator/creator.py:49]: do you need the print output for something specific?

File: fontforge_font_creator/yaml_generator.py [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
 import unicodedata
+from xml.etree import ElementTree
 
+UNICODE_MAX = 0x10FFFF  # last chr of plane 16 (plane 0x10)
 
 class UnicodeRanger:
-    def __init__(self, start, stop):
+    def __init__(self, start, stop=UNICODE_MAX):
+        assert start >= 0
+        assert stop <= UNICODE_MAX
```
Comment (hixi) [inline fontforge_font_creator/yaml_generator.py:11]: :+1:

File: fontforge_font_creator/yaml_generator.py [hunk 3]
```diff
@@ -42,18 +46,28 @@ def __next__(self):
...
 
 count = 0
+already_done = set()
 for svg in svgs:
-    print("    \"{:#06x}\":".format(r.__next__()), file=oufile_writer)
-    print("      filename: \"{}\"".format(svg), file=oufile_writer)
-    count += 1
+    tree = ElementTree.parse(os.path.join(svg_dir, svg))
+    for path_el in tree.findall('.//svg:path', ns):
+        h = hash(path_el.get('d'))
```
Comment (hixi) [inline fontforge_font_creator/yaml_generator.py:66]: good idea to use the hash :-)

File: fontforge_font_creator/creator.py (hunk 1 above)
Comment (das-g) [inline fontforge_font_creator/creator.py:49]: just progress feedback & debugging help

It as already there, I just extended it with another value. If you prefer, we can remove it.

Comment (hixi) [inline fontforge_font_creator/creator.py:49]: No, I actually think this makes sense :-)

**Pull request merged**